boa/src/syntax/lexer/comment.rs

```diff
@@ -90,3 +90,33 @@ impl<R> Tokenizer<R> for MultiLineComment {
         ))
     }
 }
+
+pub(super) struct Hashbang;
```

Comment:
I would rename this to `HashbangComment` to match the spec.

And add some documentation like with `MultiLineComment`.

Reply:
The HashbangComment makes plenty of sense. I am a bit curious about those tests if it's due to consuming the next_byte in the hashbang.lex peek once it's found the Line Separator (which would be the carriage return/new line/etc.). My thinking was that consuming so would basically start the parser off at first character of the second row, rather than the last character of the first.